.github/workflows/ci.yaml

```
@@ -185,68 +176,63 @@ jobs:
           ref: ${{ github.head_ref }}
       - name: Fetch full git history
         run: git fetch --prune --unshallow
-      - uses: conda-incubator/setup-miniconda@v2
-        with:
-          python-version: ${{ matrix.PYTHON_VERSION }}
-          miniforge-variant: Mambaforge
-          miniforge-version: 4.11.0-2
-          use-mamba: true
-          environment-file: environment.yml
-          activate-environment: datajudge
-      - name: Run Integration Tests
-        shell: bash -l {0}
-        run: |
-          flit install -s
-          pytest --cov=datajudge --cov-report=xml --cov-append --backend=db2 tests/integration
-      - name: Generate code coverage report
-        uses: codecov/codecov-action@v3.1.1
-        with:
-          file: ./coverage.xml
+      - name: Setup conda
+        uses: mamba-org/provision-with-micromamba@v15
+        with:
+          extra-specs: |
+            python=${{ matrix.PYTHON_VERSION }}
+            sqlalchemy${{ matrix.SA_VERSION }}
+            ibm_db
+            ibm_db_sa
+      - name: Run tests
+        uses: ./.github/actions/pytest
+        with:
+          backend: db2
+          args: tests/integration
 
 
-  linux-integration_tests-snowflake:
-    name: "Linux - integration tests - Python ${{ matrix.PYTHON_VERSION }} - Snowflake"
+  snowflake:
+    if: ${{ contains(github.event.pull_request.labels.*.name, 'snowflake') || contains(github.event.pull_request.labels.*.name, 'ready') }}
+    name: "Python ${{ matrix.PYTHON_VERSION }} - Snowflake - SA ${{ matrix.SA_VERSION }}"
     runs-on: ubuntu-latest
+    env:
+      SNOWFLAKE_USER: DATAJUDGE
+      SNOWFLAKE_PASSWORD: ${{ secrets.SNOWFLAKE_PASSWORD }}
+      SNOWFLAKE_ACCOUNT: ${{ secrets.SNOWFLAKE_ACCOUNT }}
     strategy:
       fail-fast: false
       matrix:
-        PYTHON_VERSION: [ '3.8' ]
+        PYTHON_VERSION: [ '3.9' ]
```

```suggestion
        PYTHON_VERSION: [ '3.8' ]
```
Let's keep 3.8 here still.